internal/ibmcloud-janitor/resources/vpc_lbs.go

```diff
@@ -99,6 +101,37 @@ func (VPCLoadBalancer) cleanup(options *CleanupOptions) error {
 	return nil
 }
 
+func shouldSkipLoadBalancer(lb vpcv1.LoadBalancer, client *IBMVPCClient, errs *[]error) bool {
+	if client.VPCID == "" {
+		// No VPC ID is provided, check if LB's resource group matches the ID passed in user-data
+		return lb.ResourceGroup == nil || *lb.ResourceGroup.ID != client.ResourceGroupID
+	}
+
+	if len(lb.Subnets) == 0 {
+		// LB does not have attached subnets, can proceed with deletion
+		resourceLogger.WithField("name", *lb.Name).Warn("load balancer has no attached subnets, assuming stale and proceed with deletion")
+		return false
+	}
+
+	// Check if subnets belong to the VPC using ID passed in user-data. If yes, skip deletion of LB
+	for _, subnetRef := range lb.Subnets {
+		subnet, _, err := client.GetSubnet(&vpcv1.GetSubnetOptions{ID: subnetRef.ID})
+		if err != nil {
+			resourceLogger.WithFields(logrus.Fields{
+				"name":      *lb.Name,
+				"subnet_id": subnetRef.ID,
+			}).Error("failed to get subnet details")
+			*errs = append(*errs, err)
+			continue
+		}
+		if subnet.VPC != nil && *subnet.VPC.ID == client.VPCID {
+			return false
+		}
+	}
+
+	return true
+}
```

Comment:
```suggestion
	// Check if subnets belong to the VPC using ID passed in user-data. If yes, skip deletion of LB
	for _, subnetRef := range lb.Subnets {
		subnet, _, err := client.GetSubnet(&vpcv1.GetSubnetOptions{ID: subnetRef.ID})
		if err != nil {
			resourceLogger.WithFields(logrus.Fields{
				"name":      *lb.Name,
				"subnet_id": subnetRef.ID,
			}).Error("failed to get subnet details")
			*errs = append(*errs, err)
			continue
		}
		if subnet.VPC != nil && *subnet.VPC.ID == client.VPCID {
			return true
		}
	}

        resourceLogger.WithField("name", *lb.Name).Warn("load balancer has no attached subnets, assuming stale and proceed with deletion")
	return false
}
```

Can this be simlified as above?

Reply:
@sudharshanibm ^^^